### PR TITLE
Random missions fixes

### LINKED ISSUE #65313: Current game keeps repeating "invalid beta talker" issue despite no NPCs being in range
### Describe the bug

See title.

### Attach save file

[Cascade-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/11330982/Cascade-trimmed.tar.gz)

### Steps to reproduce

Just walking around triggers the error.

### Expected behavior

I guess, not seeing this constant error?

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19043.1466 (21H1)
- Game Version: 01203da [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Bombastic Perks [bombastic_perks],
    Mythical Martial Arts [MMA],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Portal Storms Ignore NPCs [personal_portal_storms],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

### Additional context

 DEBUG    : Excessive error repetition detected.  Please file a bug report at https://github.com/CleverRaven/Cataclysm-DDA/issues
            Tried to use an invalid beta talker.

 FUNCTION : actor
 FILE     : D:\a\Cataclysm-DDA\Cataclysm-DDA\src\npctalk.cpp
 LINE     : 1799
 VERSION  : 01203da

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Bugfixes "fix random missions"

#### Purpose of change
resolves #65313 fixes #62640

#### Describe the solution
I added process EOCs to mutations, something I wanted anyway.

There were a bunch of ways to do this EOC workflow right, but since I already had to clean up the code, I may as well do it the cleanest way.

Made offer_mission try to assume that the alpha talker is an NPC if no beta is involved. 

#### Testing
Tested in game with a migo tower. Works the captives get assigned missions.

BUT I'm removing this cause it breaks captives 😪

### CODE REVIEW

File: data/json/effect_on_condition.json [hunk 1]
```diff
@@ -505,128 +505,126 @@
...
+              "condition": { "u_has_trait": "GETS_RANDOM_MISSION" },
               "effect": { "offer_mission": "MISSION_GET_RECORD_ACCOUNTING" }
             },
             { "const": 1 }
           ],
           [
             {
               "id": "give_MISSION_GET_INHALER",
-              "condition": { "or": [ { "npc_has_trait": "GETS_RANDOM_MISSION" }, { "npc_has_trait": "GETS_RANDOM_MISSION_NOMOVE" } ] },
+              "condition": { "or": [ { "u_has_trait": "GETS_RANDOM_MISSION" }, { "u_has_trait": "GETS_RANDOM_MISSION_NOMOVE" } ] },
```
Comment (I-am-Erk) [inline data/json/effect_on_condition.json:531]: We should add an "and" condition here to check that the NPC has the asthma trait as well.

File: data/json/effect_on_condition.json [hunk 2]
```diff
@@ -505,128 +505,126 @@
...
               "effect": { "offer_mission": "MISSION_LEARN_ABOUT_CATTAIL_JELLY" }
             },
             { "const": 1 }
           ],
           [
             {
               "id": "give_MISSION_PYROMANIAC",
-              "condition": { "npc_has_trait": "GETS_RANDOM_MISSION" },
+              "condition": { "u_has_trait": "GETS_RANDOM_MISSION" },
```
Comment (I-am-Erk) [inline data/json/effect_on_condition.json:619]: Similarly we might check that the NPC has the pyro trait for this one.

**Review by I-am-Erk — COMMENTED** [review]
Sorry for the mild bikeshedding but this is something I've wanted for a while. I can add the actual code when I get home if you want.

**Comment by Proxima-b** [comment]
I think for the mean time the random mission PR should be reverted until this issue being sorted.

**Comment by bombasticSlacks** [comment]
Our automated testing is broken, nothing C++ is getting merged, so a rollback also wouldn't. This fix has been ready for two days 🤷

**Pull request merged**